src/smt/smt_engine.cpp

```diff
@@ -3852,6 +3866,34 @@ void SmtEnginePrivate::collectSkolems(TNode n, set<TNode>& skolemSet, unordered_
   cache[n] = true;
 }
 
+void SmtEnginePrivate::computeVariablesInAssertions(
+    std::vector<Node>& assertions)
```

Comment:
This seems like it could just be a utility function rather than further polluting smt_engine.cpp

Reply:
I'm not quite sure where this should go, maybe expr/node_algorithm.h?  

My thought by putting it here was that when the non-clausal simplify preprocessing pass was moved to preprocessing/, this function would move with it.  However, if we think it should move elsewhere, I'm open to suggestions.

Reply:
Yes, expr/node_algorithm.h seems like a good place for it.  And maybe generalize it a bit to computeVariablesInNode and computeVariablesInNodeVector.

Reply:
I've added the utility function getSymbols in node_algorithm.h.  I'm holding off on doing the vector version, since I'm a bit worried that we'd have an explosion of functions in node_algorithm.h, e.g. someone else could want a unordered set version, etc. We could refactor later if this function becomes used frequently.